langchain/document_loaders/fauna.py

```diff
@@ -0,0 +1,56 @@
+from typing import List, Optional
+from langchain.docstore.document import Document
+
+from langchain.document_loaders.base import BaseLoader
+
+
+class FaunaLoader(BaseLoader):
+    def __init__(
+        self,
+        query: str,
+        page_content_field: str,
+        secrect: str,
+        metadata_fields: Optional[List[str]] = None,
+        after: Optional[str] = None,
+    ):
+        self.query = query
+        self.page_content_field = page_content_field
+        self.secrect = secrect
+        self.metadata_fields = metadata_fields
+
+    def load(self) -> List[Document]:
+        try:
+            from fauna import fql, Page
+            from fauna.client import Client
+            from fauna.encoding import QuerySuccess
+        except ImportError:
+            raise ImportError(
+                "Could not import fauna python package. "
+                "Please install it with `pip install fauna`."
+            )
+        documents = []
+        # Create Fauna Client
+        client = Client(secret=self.secrect) 
+        # Run FQL Query
+        response: QuerySuccess = client.query(fql(self.query))
+        page: Page = response.data
+        for result in page:
+            if result is not None:
+                document_dict = dict(result.items())
```

Comment:
why is this needed? what's the type of result?

Reply:
This is needed because Fauna uses a encoding to map the database types. 
https://github.com/fauna/fauna-python/blob/main/fauna/encoding/encoder.py#L49